runtime/drivers/clickhouse/olap.go

```diff
@@ -250,7 +250,57 @@ func (c *connection) InsertTableAsSelect(ctx context.Context, name, sql string,
 			LongRunning: true,
 		})
 	}
-
+	if strategy == drivers.IncrementalStrategyReplace {
+		// create temp table with the same schema
+		tempName := tempName(name)
+		// drop the temp table
+		defer func() {
+			err := c.DropTable(ctx, tempName, false)
+			if err != nil {
+				c.logger.Error("clickhouse: failed to drop temp table", zap.String("name", tempName), zap.Error(err))
+			}
+		}()
+		err := c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("CREATE TABLE %s AS %s", safeSQLName(tempName), name),
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		// insert into temp table
+		err = c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("INSERT INTO %s %s", safeSQLName(tempName), sql),
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		// list partitions from the temp table
+		res, err := c.Execute(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("SELECT DISTINCT %s FROM %s", strings.Join(key, ", "), safeSQLName(tempName)),
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		defer res.Close()
+		// iterate the partitions of the temp table
+		for res.Next() {
+			var part string
+			if err := res.Scan(&part); err != nil {
```

Comment:
The result of the below query `fmt.Sprintf("SELECT DISTINCT %s FROM %s", strings.Join(key, ", "), safeSQLName(tempName))` can't be always scanned as string.